### ISSUE TITLE
Status of HIDIVE (vrv.co)

### ISSUE THREAD

**Issue by Deathspike** [Deathspike/animesync#42]
The *HIDIVE* channel on *vrv.co* remains, unfortunately, untested.
I'm not from the USA, and *vrv.co* is determined not to accept payments without credit cards.
And we don't use credit cards in my region.

You can help by trying the download of a *vrv.co HIDIVE* series.
Are there no errors? Great, let me know!
Are there errors? Great, please send me the log!

I'd like to work with you to get support for *HIDIVE* up and running. 👍

**Comment by heptahex** [comment]
I tested animesync today. Since animesync is updated within few hours, I think this tool is not out dated and it would work. But unfortunately all of the streaming sites not working. I also tested VRV but even Crunchyroll and Funimation are not working. Since I have a premium Crunchyroll and VRV with access to HiDIVE, I'd like to help animesync to get support for HiDIVE. I also did everything exactly from the instruction. I described my workflow to download below and the log also.

Step 1
$ animesync settings credential --vrvUsername account@domain --vrvPassword mypassword
Step 2
$ animesync download https://vrv.co/watch/G6KK9WQ46/Redo-of-Healer:The-Healer-Starts-Over

Output from terminal
[11:54:46] Listening at http://127.0.0.1:45669
[11:54:46] Checking https://vrv.co/watch/G6KK9WQ46/Redo-of-Healer:The-Healer-Starts-Over
[11:54:46] Rejected https://vrv.co/watch/G6KK9WQ46/Redo-of-Healer:The-Healer-Starts-Over

For Crunchyroll and Funimation, I also did the same. Did I miss something?

**Comment by Deathspike** [comment]
Thank you for giving vrv a try!

You've used the wrong URL for the download command. It takes the series page URL instead of the episode page URL. In this case you'd use:

    animesync download https://vrv.co/watch/G6KK9WQ46/Redo-of-Healer

I'll make a note to clarify this in the documentation. I hope it works now. Please let me know 👍

**Comment by Deathspike** [comment]
Ah, here is the corrected command:

    animesync download https://vrv.co/series/GY1QK2N1R/Redo-of-Healer

**Comment by heptahex** [comment]
Oh, I get it. I gave it an another try with the show urls you provided in this repo. But still not working. Hope these logs will help.

Crunchryoll

```
#parameter
$ animesync download https://www.crunchyroll.com/a-certain-scientific-railgun

#output
[14:13:21] Listening at http://127.0.0.1:34003
[14:13:21] Checking https://www.crunchyroll.com/a-certain-scientific-railgun
[14:14:01] Rejected https://www.crunchyroll.com/a-certain-scientific-railgun
```
Funimation
```
#parameter
animesync download https://www.funimation.com/shows/overlord/

#ouput
[14:16:49] Listening at http://127.0.0.1:34997
[14:16:49] Checking https://www.funimation.com/shows/overlord/
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

TimeoutError: page.waitForNavigation: Timeout 30000ms exceeded.
=========================== logs ===========================
waiting for navigation until "domcontentloaded"
============================================================
Note: use DEBUG=pw:api environment variable and rerun to capture Playwright logs.
    at /usr/lib/node_modules/animesync/node_modules/playwright-core/lib/client/waiter.js:40:51
    at async Waiter.waitForPromise (/usr/lib/node_modules/animesync/node_modules/playwright-core/lib/client/waiter.js:48:28)
    at async /usr/lib/node_modules/animesync/node_modules/playwright-core/lib/client/frame.js:98:36
    at async Frame._wrapApiCall (/usr/lib/node_modules/animesync/node_modules/playwright-core/lib/client/channelOwner.js:72:28)
```
VRV [HiDIVE]
```
#paramter
animesync download https://vrv.co/series/GY1QK2N1R/Redo-of-Healer

#output
[14:19:50] Listening at http://127.0.0.1:38315
[14:19:50] Checking https://vrv.co/series/GY1QK2N1R/Redo-of-Healer
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "undefined".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

VRV [Crunchyroll]
```
#parameter
animesync download https://vrv.co/series/GRWMGGQ86/TONIKAWA-Over-The-Moon-For-You

#output
[14:23:21] Listening at http://127.0.0.1:44271
[14:23:21] Checking https://vrv.co/series/GRWMGGQ86/TONIKAWA-Over-The-Moon-For-You
[14:24:05] Rejected https://vrv.co/series/GRWMGGQ86/TONIKAWA-Over-The-Moon-For-You
```

**Before downloading, I double checked my credentials.**

**Comment by Deathspike** [comment]
If you're in the USA and you're logging in, or you opted-in to beta on your account, then you're using *Crunchyroll* beta. Use:

    animesync download https://beta.crunchyroll.com/series/G649J4XPY/A-Certain-Scientific-Railgun

For the others, would you mind attaching your logs? They're in:
Windows: `C:\Users\[YOURUSERNAME]\animesync\logger`
Linux: `/home/[YOURUSERNAME/.animesync/logger`
It'll contain more detailed information about problems.

**Comment by heptahex** [comment]
I'm not from US but I'm using a US VPN. It works well with youtube-dl and I can watch every US region shows from the site. So, I think it's not a VPN problem. Maybe because of my internal error? I means my installation error.

Crunchyroll
```
$ animesync download https://beta.crunchyroll.com/series/G649J4XPY/A-Certain-Scientific-Railgun

#output
[10:40:47] Listening at http://127.0.0.1:33497
[10:40:47] Checking https://beta.crunchyroll.com/series/G649J4XPY/A-Certain-Scientific-Railgun
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "undefined".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```
Funimation
```
$ animesync download https://www.funimation.com/shows/the-day-i-became-a-god

#ouput
[10:50:14] Listening at http://127.0.0.1:43505
[10:50:14] Checking https://www.funimation.com/shows/the-day-i-became-a-god
[10:50:14] Rejected https://www.funimation.com/shows/the-day-i-became-a-god
```
VRV
```
$ animesync download https://vrv.co/series/GY1QK2N1R/Redo-of-Healer

#output
[10:51:47] Listening at http://127.0.0.1:37385
[10:51:47] Checking https://vrv.co/series/GY1QK2N1R/Redo-of-Healer
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "undefined".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```
Logs
[2021-08-29.log](https://github.com/Deathspike/animesync/files/7074117/2021-08-29.log)
[2021-08-30.log](https://github.com/Deathspike/animesync/files/7074123/2021-08-30.log)

**Comment by Deathspike** [comment]
There are two issues:
1. You don't seem to have Google Chrome installed (*Crunchyroll* & *Vrv*).
2. The *Funimation* link is missing a trailing slash. It's `https://www.funimation.com/shows/the-day-i-became-a-god/`.
To solve (1) please install Google Chrome. If you already have it installed, then we'll need to find why it can't be found.

**Comment by heptahex** [comment]
Now, Crunchyroll and Funimation is working well with animesync. I'm sorry for not reading the doc properly. I misunderstood and installed Chromium instead of Google Chrome. After Google Chrome installation, both Crunchyroll and Funimation is working well. But unfortunately VRV and Crunchyroll beta is not working. For Crunchyroll, I tested the same show in both beta account and normal account. Both Crunchyroll and HiDIVE on VRV is not working. And I also attached new log to help you.

Crunchyroll (Normal) - Working
```
$ animesync add https://www.crunchyroll.com/onee-chan-ga-kita

#output
[15:37:40] Listening at http://127.0.0.1:38393
[15:37:40] Checking https://www.crunchyroll.com/onee-chan-ga-kita
[15:37:52] Updating Onee-chan ga Kita 01 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 02 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 03 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 04 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 05 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 06 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 07 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 08 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 09 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 10 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 11 [AnimeSync]
[15:37:52] Updating Onee-chan ga Kita 12 [AnimeSync]
```
Crunchyroll (Beta) - Not working
```
$ animesync add https://beta.crunchyroll.com/series/GRDQVM3KY/Onee-chan-ga-Kita

#output
[16:20:31] Listening at http://127.0.0.1:36815
[16:20:31] Checking https://beta.crunchyroll.com/series/GRDQVM3KY/Onee-chan-ga-Kita
[16:21:15] Rejected https://beta.crunchyroll.com/series/GRDQVM3KY/Onee-chan-ga-Kita
```
Funimation - Working
```
$ animesync add https://www.funimation.com/shows/combatants-will-be-dispatched/

#output
[16:25:33] Listening at http://127.0.0.1:33327
[16:25:33] Checking https://www.funimation.com/shows/combatants-will-be-dispatched/
[16:26:30] Updating Combatants Will Be Dispatched! S1 01 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 02 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 03 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 04 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 05 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 06 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 07 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 08 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 09 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 10 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 11 [AnimeSync]
[16:26:30] Updating Combatants Will Be Dispatched! S1 12 [AnimeSync]
```
VRV (Crunchyroll)
```
$ animesync add https://vrv.co/series/GRDQVM3KY/Onee-chan-ga-Kita

#output
[16:29:20] Listening at http://127.0.0.1:37099
[16:29:20] Checking https://vrv.co/series/GRDQVM3KY/Onee-chan-ga-Kita
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "undefined".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```
Crunchyroll (HiDIVE)
```
$ animesync add https://vrv.co/series/GY1QK2N1R/Redo-of-Healer

#output
[16:31:56] Listening at http://127.0.0.1:37643
[16:31:56] Checking https://vrv.co/series/GY1QK2N1R/Redo-of-Healer
[16:32:29] Rejected https://vrv.co/series/GY1QK2N1R/Redo-of-Healer
```

And I would like to request some features which is absolutely required. It seems animsync can only download shows in sequential and I can't choose specific episode from a show. It's painful when the show have a lot of episodes. If I want to download current One Piece episode, I have to download the whole show and it is impossible to download 900+ episodes and I only download and watch 10 episodes per once. One more thing I can't choose the episode quality and specific subtitles in animesync and can't choose dub language in Funimation. For VRV(HiDIVE) I'd like to help you with my best and if it's possible I'd like to contact somewhere private not on github. Thanks.

Logs
[2021-08-31.log](https://github.com/Deathspike/animesync/files/7083040/2021-08-31.log)

**Comment by Deathspike** [comment]
There are a bunch of questions and issues mixed, so I'll try breaking it down one by one.

First, you can't switch between Crunchyroll normal and Crunchyroll beta without nuking the whole animesync `chrome-data` folder. Please make sure you pick one and stick to that one (according to your account settings/region/VPN). Eventually Crunchyroll normal will be deprecated and just the beta handler will remain, but the beta website is awfully unstable right now, so pick your poison.

As for filtering which episodes to download, you can run with the `--skipDownload` arg. For example, `animesync download --skipDownload https://www.crunchyroll.com/one-piece`. This will create an `.animesync` directory with empty files representing each episode. Those won't be downloaded when you run `animesync download` again. Future episodes will be downloaded. If you delete one of these empty files and run `animesync download` again, that one specific episode will be downloaded. This is effectively a filtering system.

As for quality, subtitle and dub selection; truth be told, I don't really care. It has little priority for me. Anime deserves to be saved in the best quality, subtitles are so small it makes no sense to omit any, and English-dubs just sound awful to me. I have thought about the best solution, and that would be by adding params to the URL. That'll make everything work even when continuing a series; e.g. `https://provider.com/series?quality=720p&dub=eng`. Please make a new issue to discuss this, but don't expect this to be a high priority for me right now.

For contact to debug Vrv, please hit me up on Discord at Deathspike#3077.
I'd love to take a closer look at what's going on with Hidive :-)

**Comment by Deathspike** [comment]
Closed because of #57

**Comment by Oujiii** [comment]
Hidive is no longer a part of VRV, but still maintains a significantly part of the US streaming catalog. Should a new issue be opened for this? I can get a free trial account and provide it to you for testing.
